Cubical/Categories/Displayed/Cartesian.agda

```diff
@@ -0,0 +1,308 @@
+            symP (R.reind-filler _ _))
+          (cart .fst)
+      , λ g' →
+          Σ≡Prop
+            (λ _ → isOfHLevelPathP' 1 isSetHomᴰ _ _)
+            (cong fst $ cart .snd $
+              map-snd
+                (λ p → R.≡[]-rectify $ p R.≡[ _ ∙ _ ]∙ R.reind-filler _ _)
+                g')
+      where
+        cart : isContr (Σ _ _)
+        cart =
+          cleavage σ _ .snd .snd _ .equiv-proof $
+            R.reind (B.⋆IdL _ ∙ sym (B.⋆IdR _)) (f ⋆ᴰ cleavage σ _ .snd .fst)
+
+    substitutionFunctor : Functor (verticalCategory Cᴰ a) (verticalCategory Cᴰ b)
+    substitutionFunctor .F-ob c = cleavage σ c .fst
+    substitutionFunctor .F-hom f = substituteArrow f .fst .fst
+    substitutionFunctor .F-id {x} = cong fst $
+      substituteArrow (verticalCategory Cᴰ a .Category.id) .snd $
+        verticalCategory Cᴰ b .Category.id
+      , R.≡[]-rectify
+          ((refl R.≡[ refl ⋆ refl ]⋆ symP (R.reind-filler refl idᴰ))
+             R.≡[ cong₂ B._⋆_ refl refl ∙ _ ]∙
+           ⋆IdRᴰ (cleavage σ x .snd .fst)
+             R.≡[ B.⋆IdR σ ∙ _ ]∙
+           symP (⋆IdLᴰ (cleavage σ x .snd .fst))
+             R.≡[ sym (B.⋆IdL σ) ∙ _ ]∙
+           (R.reind-filler refl idᴰ R.≡[ refl ⋆ refl ]⋆ refl))
+    substitutionFunctor .F-seq {x} {y} {z} f g = cong fst $
+      substituteArrow (verticalCategory Cᴰ a .Category._⋆_ f g) .snd $
+        verticalCategory Cᴰ b .Category._⋆_ (stepf .fst) (stepg .fst)
+      , R.≡[]-rectify
+          ((refl
+              R.≡[ refl ⋆ sym (B.⋆IdL B.id) ]⋆
+            R.reind-filler-sym (sym $ B.⋆IdL B.id) (stepf .fst ⋆ᴰ stepg .fst))
+             R.≡[ cong₂ B._⋆_ refl (sym $ B.⋆IdL B.id) ∙ _ ]∙
+           symP (⋆Assocᴰ (cleavage σ x .snd .fst) (stepf .fst) (stepg .fst))
+             R.≡[ sym (B.⋆Assoc σ B.id B.id) ∙ _ ]∙
+           (stepf .snd R.≡[ (B.⋆IdR σ ∙ sym (B.⋆IdL σ)) ⋆ refl ]⋆ refl)
+             R.≡[ cong₂ B._⋆_ (B.⋆IdR σ ∙ sym (B.⋆IdL σ)) refl ∙ _ ]∙
+           ⋆Assocᴰ f (cleavage σ y .snd .fst) (stepg .fst)
+             R.≡[ B.⋆Assoc B.id σ B.id ∙ _ ]∙
+           (refl R.≡[ refl ⋆ (B.⋆IdR σ ∙ sym (B.⋆IdL σ)) ]⋆ stepg .snd)
+             R.≡[ cong₂ B._⋆_ refl (B.⋆IdR σ ∙ sym (B.⋆IdL σ)) ∙ _ ]∙
+           symP (⋆Assocᴰ f g (cleavage σ z .snd .fst))
+             R.≡[ sym (B.⋆Assoc B.id B.id σ) ∙ cong₂ B._⋆_ (B.⋆IdL B.id) refl ]∙
+           (R.reind-filler (B.⋆IdL B.id) (f ⋆ᴰ g) R.≡[ B.⋆IdL B.id ⋆ refl ]⋆ refl))
+      where
+        stepf :
+          Σ[ ϕ ∈ verticalCategory Cᴰ b [ cleavage σ x .fst , cleavage σ y .fst ] ]
+            (cleavage σ x .snd .fst ⋆ᴰ ϕ
+               ≡[ B.⋆IdR σ ∙ sym (B.⋆IdL σ) ]
+             f ⋆ᴰ cleavage σ y .snd .fst)
+        stepf = substituteArrow f .fst
+        stepg :
+          Σ[ ϕ ∈ verticalCategory Cᴰ b [ cleavage σ y .fst , cleavage σ z .fst ] ]
+            (cleavage σ y .snd .fst ⋆ᴰ ϕ
+               ≡[ B.⋆IdR σ ∙ sym (B.⋆IdL σ) ]
+             g ⋆ᴰ cleavage σ z .snd .fst)
+        stepg = substituteArrow g .fst
```

Comment:
any reason for writing out the type signatures here?

Reply:
None :) I just removed them